client/modules/IDE/components/CollectionList/CollectionList.jsx

```diff
@@ -22,133 +22,135 @@ import CollectionListRow from './CollectionListRow';
 import ArrowUpIcon from '../../../../images/sort-arrow-up.svg';
 import ArrowDownIcon from '../../../../images/sort-arrow-down.svg';
 
-class CollectionList extends React.Component {
-  constructor(props) {
-    super(props);
+const CollectionList = ({
+  user,
+  projectId,
+  getCollections,
+  getProject,
+  collections,
+  username: propsUsername,
+  loading,
+  toggleDirectionForField,
+  resetSorting,
+  sorting,
+  project,
+  mobile
+}) => {
+  const { t } = useTranslation();
+  const [hasLoadedData, setHasLoadedData] = useState(false);
+  const [
+    addingSketchesToCollectionId,
+    setAddingSketchesToCollectionId
+  ] = useState(null);
 
-    if (props.projectId) {
-      props.getProject(props.projectId);
+  useEffect(() => {
+    if (projectId) {
+      getProject(projectId);
     }
+    getCollections(propsUsername || user.username);
+    resetSorting();
+  }, [
+    projectId,
```

Comment:
same here, since its from the constructor do we need the dependency array?

Reply:
If the value of `propsUsername` were to change then we would want to redo the fetch with the new username.  So you need that as a dependency.

If the `projectId` were to change then you would want to redo the project fetch, but not the collection fetch.  So I would have multiple `useEffect` hooks with separate dependencies.

```
  const username = propsUsername || user.username;
  
  useEffect(() => {
    if (projectId) {
      getProject(projectId);
    }
  }, [projectId]);
  
  useEffect(() => {  
    getCollections(username);
  }, [username]);
  
 useEffect(() => {
   resetSorting();
 }, []);
```

I'm not sure what you mean by "from the constructor"?  In practice, with the way that the app is set up, the value of `propsUsername` probably doesn't change without a total unmount and remount because we never link from one user profile to another.  Maybe if you've viewing another user's collection page and you click the collections link from your user dropdown?  That's the sort of situation where there could be bugs if we don't have the right dependencies.